src/linear_algebra/tensor_product.lean

```diff
@@ -356,14 +356,23 @@ theorem smul.aux_of {R' : Type*} [has_scalar R' M] (r : R') (m : M) (n : N) :
   smul.aux r (free_add_monoid.of (m, n)) = (r • m) ⊗ₜ[R] n :=
 rfl
 
-variables [smul_comm_class R R' M] [smul_comm_class R R' N]
-variables [smul_comm_class R R'' M] [smul_comm_class R R'' N]
+variables [smul_comm_class R R' M]
+variables [smul_comm_class R R'' M]
 
--- Most of the time we want the instance below this one, which is easier for typeclass resolution
--- to find. The `unused_arguments` is from one of the two comm_classes - while we only make use
--- of one, it makes sense to make the API symmetric.
-@[nolint unused_arguments]
-instance has_scalar' : has_scalar R' (M ⊗[R] N) :=
+/-- Given two modules over a commutative semiring `R`, if one of the factors carries a
+(distributive) action of a second type of scalars `R'`, which commutes with the action of `R`, then
+the tensor product (over `R`) carries an action of `R'`.
+
+This instance defines this `R'` action in the case that it is the left module which has the `R'`
+action. Two natural ways in which this situation arises are:
```

Comment:
I think it might be worth drawing attention to the fact that we have to choose between the left and right actions, and can't have both as this would cause typeclass diamonds.

Reply:
I think I'd rather keep silent on that point here as there is a common situation in which we have action on both factors, namely the tensor product of two representations, but then the action is different. E.g., for groups it's `g • (m ⊗ₜ n) = (g • m) ⊗ₜ (g • n)` so the story starts to get more complicated.

Reply:
Related remarks [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.234773.20base.20change/near/240929258)